addon/locale.js

```diff
@@ -32,29 +32,36 @@ export default class Locale {
     }
   }
 
-  getCompiledTemplate(key, count) {
+  getCompiledTemplate(fallbackChain, count) {
     if (this.translations === undefined) { this._init(); }
 
     var result;
 
-    if (count != null) {
-      const inflection = this.pluralForm(count);
-      result = this.translations[`${key}.${inflection}`];
-    }
+    for (let i = 0; i < fallbackChain.length; i++) {
+      let key = fallbackChain[i];
+      if (count != null) {
+        const inflection = this.pluralForm(count);
+        result = this.translations[`${key}.${inflection}`];
+      }
 
-    if (result == null) {
-      result = this.translations[key];
-    }
+      if (result == null) {
+        result = this.translations[key];
+      }
+
+      if (Ember.typeOf(result) === 'string') {
+        result = this._compileTemplate(key, result);
+      }
 
-    if (Ember.typeOf(result) === 'string') {
-      result = this._compileTemplate(key, result);
+      if (result) {
+        break;
+      }
     }
 
     if (result == null) {
-      result = this._defineMissingTranslationTemplate(key);
+      result = this._defineMissingTranslationTemplate(fallbackChain[0]);
     }
 
-    Ember.assert(`Template for ${key} in ${this.id} is not a function`, Ember.typeOf(result) === 'function');
+    Ember.assert(`Template for ${fallbackChain[0]} in ${this.id} is not a function`, Ember.typeOf(result) === 'function');
```

Comment:
This might be confusing, since it could be one of the fallbacks that's wrong. I'm wondering whether we should save off the key of the translation we found so we can use it here.